python/sdk/setup.py

```diff
@@ -22,49 +22,11 @@
     "merlin.version", os.path.join("merlin", "version.py")
 ).VERSION
 
-REQUIRES = [
-    "boto3>=1.9.84",
-    "caraml-upi-protos>=0.3.1",
-    "certifi>=2017.4.17",
-    "Click>=7.0,<8.1.4",
-    "cloudpickle==2.0.0",  # used by mlflow
-    "cookiecutter>=1.7.2",
-    "dataclasses-json>=0.5.2",  # allow Flyte version 1.2.0 or above to import Merlin SDK
-    "docker<=6.1.3",
-    "GitPython>=3.1.40",
-    "google-cloud-storage>=1.19.0",
-    "protobuf>=3.12.0,<5.0.0",  # Determined by the mlflow dependency
-    "mlflow==1.26.1",
-    "PyPrind>=2.11.2",
-    "python_dateutil>=2.5.3",
-    "PyYAML>=5.4",
-    "six>=1.10",
-    "urllib3>=1.26",
-    "numpy<=1.23.5",  # Temporary pin numpy due to https://numpy.org/doc/stable/release/1.20.0-notes.html#numpy-1-20-0-release-notes
-    "caraml-auth-google==0.0.0.post7",
-    "pydantic==2.5.3"
-]
+with open("requirements.txt") as f:
+    REQUIRE = f.read().splitlines()
```

Comment:
Refactored the listing of requirements away into a separate file, just like how it's done in the batch predictor and pyfunc server packages (see https://github.com/caraml-dev/merlin/blob/main/python/batch-predictor/setup.py#L24 for an example).